### PR TITLE
building auth

### PR DESCRIPTION
**Pull request by isrealanteneh**
I built the basic auth in this contribution, and I want you Dave to watch the code and give me more detailed information.

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "description": "Awesome project developed with TypeORM.",
+  "type": "commonjs",
+  "devDependencies": {
+    "@types/express": "^4.17.21",
+    "@types/node": "^16.11.10",
+    "ts-node": "10.9.1",
+    "typescript": "4.5.2"
+  },
+  "dependencies": {
+    "@types/jsonwebtoken": "^9.0.6",
```
Comment (dawit-kebebe) [inline backend/package.json]: I think this is supposed to be a dev dependency not production dependancy.

File: backend/src/auth.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+dotenv.config();
+
+export default async function authUser(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const cookie = req.headers.cookie;
```
Comment (dawit-kebebe) [inline backend/src/auth.middleware.ts]: isn't it better if we make it in a header authentication bearer attribute rather than the cookie ?

**Comment by dawit-kebebe** [comment]
I think it is better if you put all of the code you wrote inside index.ts inside app.ts because I have wrote some other stuff in there too. I will push the directory structure if possible, synchronize your repo before pull request. 
And also, it was better if you could write a function to verify the user token instead of only middleware. because we might need to verify the token from other places in our code too and we might need to handle the verification process differently.

**Comment by isrealanteneh** [comment]
1, i have made some changes on my package.json file to make it dev dependencies
2. change the name to app.ts
3. the verifyed user id can be access from any router by importing the `verifyToken` function from authuser middleware. so now it is not only middleware we can reuse it
4. about the token i prefer to put it in cookie than session. if doing that have issue let me know.

**Pull request merged**